lib/frame_enum.ml

```diff
@@ -0,0 +1,67 @@
+open Core_kernel.Std
+
+module type E = sig
+  type t [@@deriving enumerate]
+end
+
+module type Enum = sig
+  include E
+
+  val to_enum : t -> int
+  val of_enum : int -> t option
+  val max : int
+  val min : int
+end
+
+module type L = sig
+  include E
+  val list : (int * t) list
+end
+
+module Make_list(A : E) = struct
+  include A
+  let list = List.mapi all ~f:(fun i x -> i, x)
+end
+
+module Make_enum(A : L) : Enum with type t := A.t = struct
+  include A
+
+  let sorted = List.sort ~cmp:(fun (i,_) (j,_) -> compare i j) list
```

Comment:
1. It is better not to rely on a polymorphic comparison function in generic code. So, if you want to sort then put the comparison function into interface. It is better not to rely on it at all, the only exception if it is local to function, and will never escape the context (e.g., when you're applying a compare function to a ground type and you know that the comparison function of this ground type is well-suited for your task).

2. Are you sure that we need to sort? If we are trying to mimic the behavior of the `enum` deriver, then, as far as I know, it preserves the order of constructor declarations. 

3. An array would be a match better (more efficient and natural) representation. (a) you can perform an O(1) lookup by index, (b) you can perform an O(log N) lookup by name (though this would require the sorting, and I have a strong belief, that we do not need to sort (see point 2).